Split libtorch agnostic tests by feature version

mikaylagawarecki opened the pull request:
Tests are split into libtorch_agnostic_2_9_extension and libtorch_agnostic_2_10_extension depending on the minimum version they should compile+run in

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #168062
* #168034
* #167962
* #167804
* __->__ #167803
* #167802
* #168025

mikaylagawarecki committed:
Split libtorch agnostic tests by feature version

[ghstack-poisoned]

janeyx99 reviewed (commented):
I'm understanding this is a draft PR for now but I have a q: is it right to expect the diff between 2_9 and 2_10 is 2_9 will have fewer kernels that are not dependent on 2_10?

mikaylagawarecki committed:
Update on "Split libtorch agnostic tests by feature version"




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Split libtorch agnostic tests by feature version"




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Split libtorch agnostic tests by feature version"




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Split libtorch agnostic tests by feature version"




[ghstack-poisoned]

janeyx99 reviewed (approved):
skimmed, looks fine to me

mikaylagawarecki committed:
Update on "Split libtorch agnostic tests by feature version"

Tests are split into libtorch_agnostic_2_9_extension and libtorch_agnostic_2_10_extension depending on the minimum version they should compile+run in




[ghstack-poisoned]
